### PR TITLE
[docs] [ui] Make Pro badge style consistent

### PR DESCRIPTION
**Pull request by bharatkashyap**
- Make uppercase the standard for the "Pro" chip:
Screenshot 2024-04-18 at 3 59 49 PM

### CODE REVIEW

File: packages/toolpad-studio/src/toolpad/AppEditor/UpgradeNotification.tsx [hunk 1]
```diff
@@ -62,7 +62,7 @@ export function UpgradeChip({
         size="small"
         clickable
         label="Pro"
-        sx={sx}
+        sx={{ textTransform: 'uppercase', ...sx }}
```
Comment (Janpot) [inline packages/toolpad-studio/src/toolpad/AppEditor/UpgradeNotification.tsx:65]: Why not just?
```tsx
label="PRO"
```

**Comment by prakhargupta1** [comment]
Olivier mentioned using "Pro" and he did the changes for X repo in https://github.com/mui/mui-x/pull/12922
I think we should follow the same convention.

**Comment by bharatkashyap** [comment]
> Olivier mentioned using "Pro" and he did the changes for X repo in [mui/mui-x#12922](https://github.com/mui/mui-x/pull/12922) I think we should follow the same convention.

More than nomenclature this is a typography question, since text inside Chips is usually capitalised, as in on our docs as well:

Screenshot 2024-04-29 at 5 02 45 PM

We can continue to use "Pro" (instead of "pro" or "PRO") in the documentation text.

File: packages/toolpad-studio/src/toolpad/AppEditor/UpgradeNotification.tsx [hunk 2]
```diff
@@ -61,7 +61,7 @@ export function UpgradeChip({
         rel="noopener"
         size="small"
         clickable
-        label="Pro"
+        label="PRO"
```
Comment (Janpot) [inline packages/toolpad-studio/src/toolpad/AppEditor/UpgradeNotification.tsx:64]:
```suggestion
        label="Pro"
```

**Review by Janpot — APPROVED** [review]
Looks great!

**Pull request merged**